lib/tpm2_auth_util.c

```diff
@@ -94,6 +97,12 @@ static bool handle_str(const char *password, TPMS_AUTH_COMMAND *auth) {
         password += STR_PREFIX_LEN;
     }
 
+    bool is_hmac_prefix = !strncmp(password, HMAC_SESSION_PREFIX,
+        HMAC_SESSION_PREFIX_LEN);
+    if (is_hmac_prefix) {
+        password += HMAC_SESSION_PREFIX_LEN;
+    }
```

Comment:
You're polluting the routine handle_str with hmac logic to re-use the code block on 109. The right way to reuse that code block is to move it to a seperate routine and call it.

You should have a routine called handle_hmac() that does this hmac prefix jump, and then call that copy routine.